### PR TITLE
Expose some more FUSE 7.23 flags

### PR DESCRIPTION
**Pull request by yallop**
Expose some more FUSE 7.23 flags, and switch multi-version strategy: `Profuse_7_23` is now an independent implementation rather than an extension of `Profuse_7_8`.

### CODE REVIEW

File: lib/profuse_7_23.mli [hunk 1]
```diff
@@ -106,6 +106,13 @@ module Types : sig
   module Out : sig
 
     module Hdr : sig
+      module Notify_code :
+      sig
+        val fuse_notify_poll : int32
+        val fuse_notify_inval_inode : int32
+        val fuse_notify_inval_entry : int32
```
Comment (dsheets) [inline lib/profuse_7_23.mli]: Most other instances of alternate type codes are exposed via variants.

File: lib/profuse_7_8.mli [hunk 2]
```diff
@@ -122,6 +122,12 @@ module Types : sig
     end
 
     module Open : sig
+      module Flags :
+      sig
+        val fopen_direct_io : Unsigned.uint32
+        val fopen_keep_cache : Unsigned.uint32
```
Comment (dsheets) [inline lib/profuse_7_8.mli]: Why not variants?

**Comment by dsheets** [comment]
This looks really nice. My primary quibble with it is the use of integers rather than variants. Maybe that makes sense and is a good idea for GC and low-level protocol reasons. I do like real types and pattern matching, though. What's the rationale for exposing integers directly?

File: lib/profuse_7_23.mli (hunk 1 above)
Comment (yallop) [inline lib/profuse_7_23.mli]: Good point.  I've switched to variants in 2a2219a0.

File: lib/profuse_7_8.mli (hunk 2 above)
Comment (yallop) [inline lib/profuse_7_8.mli]: 681cc33 changes the flags to a polymorphic variant

**Comment by dsheets** [comment]
Great! Thank you!

**Pull request merged**